contracts/root/withdrawManager/WithdrawManager.sol

```diff
@@ -272,54 +133,102 @@ contract WithdrawManager is WithdrawManagerStorage /* , IWithdrawManager */ {
     require(ownerExits[key] == 0, "EXIT_ALREADY_IN_PROGRESS");
 
     // Calculate priority.
-    uint256 exitableAt = Math.max(_createdAt + 2 weeks, block.timestamp + 1 weeks);
+    uint256 exitableAt = Math.max(now + 2 weeks, block.timestamp + 1 weeks);
 
     PriorityQueue queue = PriorityQueue(exitsQueues[_exitObject.token]);
-    queue.insert(exitableAt, _exitId);
+    queue.insert(exitableAt, priority);
 
     // create NFT for exit UTXO
     // @todo
-    ExitNFT(exitNFTContract).mint(_exitObject.owner, _exitId);
-    exits[_exitId] = _exitObject;
+    ExitNFT(exitNFTContract).mint(_exitObject.owner, priority);
+    exits[priority] = _exitObject;
 
     // set current exit
-    ownerExits[key] = _exitId;
+    ownerExits[key] = priority;
 
     // emit exit started event
-    emit ExitStarted(_exitObject.owner, _exitId, _exitObject.token, _exitObject.receiptAmountOrNFTId);
+    emit ExitStarted(exitor, priority, rootToken, exitAmountOrTokenId, burnt);
+  }
+
+  function addInput(uint256 exitId, uint256 age, address signer) external {
+    PlasmaExit storage exitObject = exits[exitId];
```

Comment:
this is really cool, I am wondering if we are doing any validation before adding new inputs?

Reply:
Aah, great catch 👏 
I meant to ACL it on `isPredicateAuthorized`.